software/deploy/deploy.go

```diff
@@ -164,6 +165,33 @@ func validateRuntimeVersion(houstonClient houston.ClientInterface, tag string, d
 	return nil
 }
 
+func updateDeploymentImageAPICall(houstonClient houston.ClientInterface, imageName, releaseName, airflowVersion, runtimeVersion string) (interface{}, error) {
```

Comment:
what's the value addition by having this extra function, that internally just calls the Houston API?

Reply:
It's being called from 2 places. Separated it out so that the API call logic remains here for extensibility and DRY.

Reply:
there is hardly anything to DRY in here 🤷‍♂️ (it is calling a single function in here), and I am not sure how we would extend it in the future.
I don't think we are gaining anything out of this extra function at the moment.

Reply:
Advantages:
1. Code Reusability
It centralizes the request creation logic
Reduces duplication
Makes the code more DRY

2. Improved Readability
Separates concerns
Makes the calling code cleaner

3. Flexibility
Easier to modify request creation in one place
Supports potential future variations (addition of new request parameters and headers in future)

Reply:
Since we are on this topic, think in terms of redundancy and maintainability
- It introduces an unnecessary layer of abstraction. The function merely wraps another function call `(houston.Call(houstonClient.UpdateDeploymentImage)(req))` without adding any significant value or functionality. So if you think about it, that also makes it not so flexible, because if we need to modify the request, then we would need to update this function signature and then both the caller function as well, since everything is just passed on by this function.
- It adds complexity to the codebase. Future maintainers must navigate through an additional function call to understand the logic. So in fact it reduces the reliability of the file. Separation of concern does not come into play here since this is just a wrapper function.